x/rollapp/keeper/rollapp.go

```diff
@@ -3,11 +3,104 @@ package keeper
 import (
 	"fmt"
 
+	errorsmod "cosmossdk.io/errors"
 	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+
 	"github.com/dymensionxyz/dymension/v3/x/rollapp/types"
 )
 
+func (k Keeper) RegisterRollapp(ctx sdk.Context, rollapp types.Rollapp) error {
+	if err := rollapp.ValidateBasic(); err != nil {
+		return fmt.Errorf("validate rollapp: %w", err)
+	}
+
+	if err := k.checkIfRollappExists(ctx, rollapp.RollappId, rollapp.Alias); err != nil {
+		return err
+	}
+
+	if err := k.checkIfInitialSequencerAddressTaken(ctx, rollapp.InitialSequencerAddress); err != nil {
+		return fmt.Errorf("check if initial sequencer address taken: %w", err)
+	}
+
+	if err := k.checkIfBech32PrefixTaken(ctx, rollapp.Bech32Prefix); err != nil {
+		return fmt.Errorf("check if bech32 prefix taken: %w", err)
+	}
+
+	creator, _ := sdk.AccAddressFromBech32(rollapp.Creator)
+	registrationFee := sdk.NewCoins(k.RegistrationFee(ctx))
+
+	if err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, creator, types.ModuleName, registrationFee); err != nil {
+		return errorsmod.Wrap(types.ErrFeePayment, err.Error())
+	}
+
+	if err := k.bankKeeper.BurnCoins(ctx, types.ModuleName, registrationFee); err != nil {
+		return fmt.Errorf("burn coins: %w", err)
+	}
+
+	k.SetRollapp(ctx, rollapp)
+
+	// Emit event
+	if err := ctx.EventManager().EmitTypedEvent(&rollapp); err != nil {
+		return fmt.Errorf("emit event: %w", err)
+	}
+
+	return nil
+}
+
+func (k Keeper) checkIfRollappExists(ctx sdk.Context, id, alias string) error {
+	rollappId, err := types.NewChainID(id)
+	if err != nil {
+		return err
+	}
+	// check to see if the RollappId has been registered before
+	if _, isFound := k.GetRollapp(ctx, rollappId.GetChainID()); isFound {
+		return types.ErrRollappExists
+	}
+
+	if _, isFound := k.GetRollappByAlias(ctx, alias); isFound {
```

Comment:
what the expected behaviour for rollapp hard fork? the alias needs to be changed as well?

Reply:
this is being researched in another issue, no need to tackle it here